app/modules/firebase-db.js

```diff
@@ -54,20 +56,45 @@ export function saveProfileToFb(data) {
       .catch((err) => {
         throw new Error(`Error getting cat profile: ${err}`);
       })
-    );
+  );
 }
 
-export function fetchCatsFromFb(uid) {
+function merge(cats, snaps) {
+  return _.map(cats, (cat, index) => _.assign(
+    {},
+    cat,
+    snaps[index] || {}
+  ));
```

Comment:
I think with object rest/spread, the right side of the arrow fn could just be 
```
{
  ...cat,
  ...snaps[index]
}
```
Just an FYI. I wouldn't advocate that change here; the explicit fallback to empty object is probably more clear.